traffic_ops/traffic_ops_golang/profile/profiles.go

```diff
@@ -149,7 +149,7 @@ func (prof *TOProfile) Read(h http.Header, useIMS bool) ([]interface{}, error, e
 
 	rows, err := prof.ReqInfo.Tx.NamedQuery(query, queryValues)
 	if err != nil {
-		return nil, nil, errors.New("profile read querying: " + err.Error()), http.StatusInternalServerError, nil
+		return nil, err, errors.New("profile read querying: " + err.Error()), http.StatusInternalServerError, nil
```

Comment:
As I mentioned in the previous comment, this should just stay the same as it was before.
You could add another validation for the param being an int.

Reply:
done.